### PR TITLE
Prototype of new architecture, without work manager API.

### PR DESCRIPTION
**Pull request by rtibbles**
Starts the server inside the application process.
Defers only task workers to a foregrounded service.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
 from android_utils import start_service
 from jnius import autoclass
 from kolibri.main import enable_plugin
 from kolibri.plugins.app.utils import interface
 from kolibri.utils.cli import initialize
-from kolibri.utils.server import _read_pid_file
-from kolibri.utils.server import PID_FILE
-from kolibri.utils.server import STATUS_RUNNING
-from kolibri.utils.server import wait_for_status
+from kolibri.utils.server import BaseKolibriProcessBus
```
Comment (dylanmccall) [inline src/main.py:10]: This will fail awkwardly if we're using Kolibri < 0.16. Should we add an extra check to the Makefile so this gets caught early?

Comment (rtibbles) [reply src/main.py:10]: Can do - we've mostly relied on tagging and versioning of the installer repos to keep that in check, but doesn't hurt to add extra.

At the very least, we can update the README.

**Comment by dylanmccall** [comment]
This looks pretty good to me and I have a branch merging it with some of the stuff we've glued together downstream: endlessm/kolibri-installer-android#20. I also add a bit of refactoring, and my thought is I'll split that into a different pull request once we have this sorted out.

Using the newest build of Kolibri from the `develop` branch, the workers service seems to be doing its thing: worker-related stuff appears in logcat without any obvious errors. With that said, it does run into an error importing content, but I think that's just a problem in Kolibri, not here. (It gets a 404 requesting `/device/api/device_channel/000409f81dbe5d1ba67101cb9fed4530/?include_fields=total_resources%2Ctotal_file_size%2Con_device_resources%2Con_device_file_size%2Cnew_resource_count%2Cnew_resource_total_size&1652997798744=1652997798744`, and then throws an exception here: https://github.com/learningequality/kolibri/blob/1fd1fc08c92eae64634126a757c9e1af442870c2/kolibri/core/assets/src/api-resources/task.js#L14-L16).

**Comment by rtibbles** [comment]
Oh yes, absolute latest develop is completely broken on the frontend for tasks - because of some of the Task API refactoring that I alluded to earlier. I am fixing it as we speak.

**Comment by rtibbles** [comment]
Going to merge this for now, we can make more updates as we go.

**Pull request merged**